express_relay/examples/easy_lend/contracts/EasyLend.sol

```diff
@@ -0,0 +1,341 @@
+// Copyright (C) 2024 Lavra Holdings Limited - All Rights Reserved
+pragma solidity ^0.8.13;
+
+import "./EasyLendStructs.sol";
+import "./EasyLendErrors.sol";
+import "forge-std/StdMath.sol";
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+
+import "@pythnetwork/pyth-sdk-solidity/PythStructs.sol";
+import "@pythnetwork/pyth-sdk-solidity/IPyth.sol";
+import "@pythnetwork/express-relay-sdk-solidity/IExpressRelayFeeReceiver.sol";
+import "@pythnetwork/express-relay-sdk-solidity/IExpressRelay.sol";
+
+contract EasyLend is IExpressRelayFeeReceiver {
+    using SafeERC20 for IERC20;
+
+    event VaultReceivedETH(address sender, uint256 amount, bytes permissionKey);
+
+    uint256 _nVaults;
+    address public immutable expressRelay;
+    mapping(uint256 => Vault) _vaults;
+    address _oracle;
+
+    /**
+     * @notice EasyLend constructor - Initializes a new token vault contract with given parameters
+     *
+     * @param expressRelayAddress: address of the express relay
+     * @param oracleAddress: address of the oracle contract
+     */
+    constructor(address expressRelayAddress, address oracleAddress) {
```

Comment:
i think we should include the recent updates to allow undercollateralized vaults. it would be good for the protocol that users see in the examples to match the behavior they see on testnet